### PR TITLE
docs: technology

### PR DESCRIPTION
**Pull request by tolerants**
closes https://spryker.atlassian.net/browse/HRZ-2146

### CODE REVIEW

File: _drafts/oryx/technology.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Oryx technologies
+
+Oryx is a frontend framework that is designed to help developers build web applications using modern technologies.
+It is built on a variety of technologies and in this article, we will take a closer look at each of these technologies and see how they contribute to the development of Oryx.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:4]: I'd emphasize that:
- Oryx uses standards wherever available and applicable. This is true for the technologies and tools
- Oryx tries to use the least necessary build tools and likes to stick to the "native" web where possible.
- Oryx aims to use JavaScript during development, but tries avoid the need to run JavaScript in the browser. Wherever possible, html and CSS are preferred to render a fast loading experience cross different screen sizes

File: _drafts/oryx/technology.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## HTML
+HTML is the backbone of the web. It is the markup language used to create web pages and applications. In Oryx, HTML is used to structure the content and layout of the application. It provides a semantic structure to the page, making it easier to understand and maintain.
+
+## CSS
+CSS is a style sheet language used to describe the visual presentation of a document written in HTML. In Oryx, CSS is used to style the components and layout of the application. It allows developers to create a consistent visual style across the application and ensure that it is responsive and accessible.
+
+## JavaScript
+JavaScript is a high-level programming language that is widely used in web development. In Oryx, JavaScript is used to add interactivity and dynamic behavior to the application. It allows developers to create complex logic and manipulate the DOM in real-time.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:13]: We can perhaps merge HTML, CSS and JS together, as it's the "web big three".  It should not have a lot of explanation.

File: _drafts/oryx/technology.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+## Web Components
+Web Components are a set of web platform APIs that allow developers to create custom, reusable, and encapsulated HTML elements. In Oryx, Web Components are used to create reusable UI components that can be used across the application. It allows developers to create complex components with minimal code duplication and provides better code organization and maintainability. At the same time it means Oryx component are compatible with any other frontend framework and can be integrated into almost any web-application.
+
+## Lit
+Lit is a lightweight library for creating web components using TypeScript and HTML templates. In Oryx, Lit is used to create reusable and encapsulated UI components that can be easily styled and extended. It provides a simple and intuitive API for creating custom elements and allows developers to write less boilerplate code.
+
+## RxJS
+RxJS is a reactive programming library for JavaScript. In Oryx, RxJS is used to handle asynchronous operations. It provides a declarative way to handle events and data streams, making it easier to manage complex data flows and state transitions.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:25]: We could add that 
- RxJS is used in the service layer of Oryx to provide a reactive system that continuees to emit new values while you navigate through the (SPA) experience. 
- RxJS is not a main citizen in the component logic, as we try to hide the reactivity altogether to avoid complexity

File: _drafts/oryx/technology.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+## Vite
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:27]: Let's make a note that Vite is our favorite, but it's optional.

File: _drafts/oryx/technology.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# Oryx technologies
+
+Oryx is a frontend framework that is designed to help developers build web applications using modern technologies.
+It is built on a variety of technologies and in this article, we will take a closer look at each of these technologies and see how they contribute to the development of Oryx.
+
+## HTML
+HTML is the backbone of the web. It is the markup language used to create web pages and applications. In Oryx, HTML is used to structure the content and layout of the application. It provides a semantic structure to the page, making it easier to understand and maintain.
+
+## CSS
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:9]: For CSS we could add here:
- we use standard CSS, no BEM, no preprocessors, no postCSS. (we'll add more on CSS in a separate section,we can link it together later)

File: _drafts/oryx/technology.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+CSS is a style sheet language used to describe the visual presentation of a document written in HTML. In Oryx, CSS is used to style the components and layout of the application. It allows developers to create a consistent visual style across the application and ensure that it is responsive and accessible.
+
+## JavaScript
+JavaScript is a high-level programming language that is widely used in web development. In Oryx, JavaScript is used to add interactivity and dynamic behavior to the application. It allows developers to create complex logic and manipulate the DOM in real-time.
+
+## TypeScript
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:15]: We could also add here that we use TS to transpile to different JS versions. Not sure how important it still is though.

File: _drafts/oryx/technology.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
+# Oryx technologies
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:1]: we _could_ think about adding some structure:

### Deployed stack:
- HTML
- CSS
- JS
- Web hosting
- CDN

### Dev tech
- Node (dev server)
- Typescript
- Vite
- RxJS

We should maybe mention here SSR?

What about explaining the why rather than the what as Andrii suggested?

File: _drafts/oryx/technology.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+## Web Components
+Web Components are a set of web platform APIs that allow developers to create custom, reusable, and encapsulated HTML elements. In Oryx, Web Components are used to create reusable UI components that can be used across the application. It allows developers to create complex components with minimal code duplication and provides better code organization and maintainability. At the same time it means Oryx component are compatible with any other frontend framework and can be integrated into almost any web-application.
+
+## Lit
+Lit is a lightweight library for creating web components using TypeScript and HTML templates. In Oryx, Lit is used to create reusable and encapsulated UI components that can be easily styled and extended. It provides a simple and intuitive API for creating custom elements and allows developers to write less boilerplate code.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:22]: I'd like to emphasis that web components can be build with any framework. We like LIT and strongly believe in it, but we can mix it with other frameworks.

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: _drafts/oryx/technology.md (hunk 2 above)
Comment (tobi-or-not-tobi) [reply _drafts/oryx/technology.md]: fair, but I can imagine we tackle that in the styling section. This is more high-level so happy with the changes you did.

File: _drafts/oryx/technology.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+## Deployed stack:
+
+### HTML & CSS & JS
+HTML, CSS, and JS are considered to be the fundamental and essential set of web technologies used for creating web pages. HTML is responsible for the structure and content of the page, CSS is used for styling and layout, and JS adds interactivity and dynamic behavior to the page. Together, these three technologies form the foundation of modern web development.
+
+### Web Components
+Web Components are a set of web platform APIs that allow developers to create custom, reusable, and encapsulated HTML elements.
+
+### Web hosting
+Oryx uses Netlify as a hosting provider, which means you can easily find basic boilerplate to deploy your site on Netlify. However, if you choose to use a different hosting provider, there may be limitations on the implementation of SSR. To make it work, you would need to adjust the SSR rendering boilerplate, which is specific to the hosting provider.
```
Comment (dunqan) [inline _drafts/oryx/technology.md:19]: Maybe rephrase it to something like "currently" or "by default"? As it sounds a bit like netlify is our preffered or final choice. @tobi-or-not-tobi ?

File: _drafts/oryx/technology.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
+# Oryx technologies
+
+Oryx is a frontend framework that is designed to help developers build web applications using modern technologies.
+When possible and appropriate, Oryx adheres to standards for both technologies and tools.
+Oryx strives to minimize the use of build tools and instead utilize the native capabilities of the web, as much as possible.
+Oryx aims to use JavaScript during development, but tries avoid the need to run JavaScript in the browser. Wherever possible, HTML and CSS are preferred to render a fast loading experience cross different screen sizes.
+
+We will examine both the deployed technology stack and the technologies utilized for development in order to gain a more comprehensive understanding of the technologies.
+
+## Deployed stack:
```
Comment (dunqan) [inline _drafts/oryx/technology.md:10]:
```suggestion
## Development stack: 
```
?

File: _drafts/oryx/technology.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+### CSS
+CSS is a style sheet language used to describe the visual presentation of a document written in HTML. In Oryx, CSS is used to style the components and layout of the application. It allows developers to create a consistent visual style across the application and ensure that it is responsive and accessible.
+
+Oryx uses standard CSS syntax and does not utilize CSS preprocessors or post-processors, such as Sass or PostCSS. 
+
+Also unlike some other CSS methodologies, such as BEM (Block, Element, Modifier), Oryx does not rely on a specific naming convention to structure its CSS classes.
+
+### JavaScript
+JavaScript is a high-level programming language that is widely used in web development. In Oryx, JavaScript is used to add interactivity and dynamic behavior to the application. It allows developers to create complex logic and manipulate the DOM in real-time.
```
Comment (dunqan) [inline _drafts/oryx/technology.md:41]: are we using JS or TS?

File: _drafts/oryx/technology.md (hunk 7 above)
Comment (dunqan) [reply _drafts/oryx/technology.md:1]: I know the approach to structure is opinionated, but I don't super like, that we mention JS in deployed and development stack. 

Different approach I could think about:

Technology
 - html, css
 - typescript
 - web components
 - lit
 - rxjs
 - nodejs
 - lambda
 - SSR
- vite

 Tooling & deployment:
- storybook
- netlify

File: _drafts/oryx/technology.md (hunk 9 above)
Comment (tolerants) [reply _drafts/oryx/technology.md:19]: Good point, added that we are currently using it.

File: _drafts/oryx/technology.md (hunk 7 above)
Comment (tolerants) [reply _drafts/oryx/technology.md:1]: We can quickly discuss it on our sync.

File: _drafts/oryx/technology.md (hunk 9 above)
Comment (tobi-or-not-tobi) [reply _drafts/oryx/technology.md:19]: We c/should actually drop the notion of Netlify entirely. There's no dependency on Netlify, but we do have a toml file and some integration with lambda which might make it opinionated

File: _drafts/oryx/technology.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+
+### Web Components
+In Oryx, Web Components are used to create reusable UI components that can be used across the application. It allows developers to create complex components with minimal code duplication and provides better code organization and maintainability. At the same time it means Oryx component are compatible with any other frontend framework and can be integrated into almost any web-application.
+
+### Lit
+Lit is a lightweight library for creating web components using TypeScript and HTML templates. In Oryx, Lit is used to create reusable and encapsulated UI components that can be easily styled and extended. It provides a simple and intuitive API for creating custom elements and allows developers to write less boilerplate code.
+
+### RxJS
+RxJS is a reactive programming library for JavaScript. In Oryx, RxJS is used to handle asynchronous operations. It provides a declarative way to handle events and data streams, making it easier to manage complex data flows and state transitions.
+RxJS is used in the service layer of Oryx to provide a reactive system that continuees to emit new values while you navigate through the (SPA) experience.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:51]:
```suggestion
RxJS is predominantly used in the service layer to provide a reactive system that emit new values while you navigate through the (SPA) experience.
```

File: _drafts/oryx/technology.md (hunk 4 above)
Comment (tobi-or-not-tobi) [reply _drafts/oryx/technology.md]: It gives the impression that we require customers to use Vite.

File: _drafts/oryx/technology.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+Server-side rendering, on the other hand, involves generating a complete HTML document on the server and then sending it to the browser. This approach provides faster initial load times, better SEO, and improved accessibility. However, it can be more complex to implement and may not be as effective for highly dynamic pages that require frequent updates.
+
+## Development stack:
+
+### HTML
+HTML is the backbone of the web. It is the markup language used to create web pages and applications. In Oryx, HTML is used to structure the content and layout of the application. It provides a semantic structure to the page, making it easier to understand and maintain.
+
+### CSS
+CSS is a style sheet language used to describe the visual presentation of a document written in HTML. In Oryx, CSS is used to style the components and layout of the application. It allows developers to create a consistent visual style across the application and ensure that it is responsive and accessible.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:34]:
```suggestion
```

This is too obvious imho
```suggestion
CSS is a style sheet language used to describe the visual presentation of a document written in HTML. In Oryx, CSS is used to style the components and layout of the application. It allows developers to create a consistent visual style across the application and ensure that it is responsive and accessible.
```

File: _drafts/oryx/technology.md [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+When possible and appropriate, Oryx adheres to standards for both technologies and tools.
+Oryx strives to minimize the use of build tools and instead utilize the native capabilities of the web, as much as possible.
+Oryx aims to use JavaScript during development, but tries avoid the need to run JavaScript in the browser. Wherever possible, HTML and CSS are preferred to render a fast loading experience cross different screen sizes.
+
+We will examine both the deployed technology stack and the technologies utilized for development in order to gain a more comprehensive understanding of the technologies.
+
+## Deployed stack:
+
+### HTML & CSS & JS
+HTML, CSS, and JS are considered to be the fundamental and essential set of web technologies used for creating web pages. HTML is responsible for the structure and content of the page, CSS is used for styling and layout, and JS adds interactivity and dynamic behavior to the page. Together, these three technologies form the foundation of modern web development.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:13]: It would actually be nice to express that:
- we try to avoid loading JS in the client unless there's a need; we gradually load snippets of JS rather than a full JS framework
- we have a CSS first approach. CSS is always preferred to render the styles and layouts, as it won't requires JS logic.

File: _drafts/oryx/technology.md [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+
+## Development stack:
+
+### HTML
+HTML is the backbone of the web. It is the markup language used to create web pages and applications. In Oryx, HTML is used to structure the content and layout of the application. It provides a semantic structure to the page, making it easier to understand and maintain.
+
+### CSS
+CSS is a style sheet language used to describe the visual presentation of a document written in HTML. In Oryx, CSS is used to style the components and layout of the application. It allows developers to create a consistent visual style across the application and ensure that it is responsive and accessible.
+
+Oryx uses standard CSS syntax and does not utilize CSS preprocessors or post-processors, such as Sass or PostCSS.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:36]: - We should also leave a not about css in js I believe. 
- Sass and PostCSS techniques are no longer used since they add little value in a modern web framework.
  - PostCSS is no longer a big thing since browsers are much more aligned
  - Sass is no longer needed since Oryx does not use a global CSS utility library; instead, we have isolated styles per component and because of that we do not want to load global css utils in each component.

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: _drafts/oryx/technology.md (hunk 15 above)
Comment (tolerants) [reply _drafts/oryx/technology.md:36]: Hm, it looks like we are going deeper and deeper into `we are not using` mode. I'm kind of agree with preprocessors, as long as they are really wide used. But why we want to add anything else we are using? I think there is no added value in mentioning we are not using PostCSS. And the same about css-in-js.

File: _drafts/oryx/technology.md [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+HTML, CSS, and JS are considered to be the fundamental and essential set of web technologies used for creating web pages. HTML is responsible for the structure and content of the page, CSS is used for styling and layout, and JS adds interactivity and dynamic behavior to the page. Together, these three technologies form the foundation of modern web development.
+
+In Oryx we prioritize CSS as the primary means of styling and layout rendering, embracing a CSS-first approach. We favor CSS over JavaScript for its inherent capability to handle styling and layout tasks without the need for additional JavaScript logic.
+
+We are  to avoid loading JS in the client unless there's a need; we gradually load snippets of JS rather than a full JS framework.
+
+In the Oryx framework, our approach is to minimize the loading of JavaScript on the client side unless it is necessary. Instead of loading a full JavaScript framework, we opt for gradually loading snippets of JavaScript as required.
+
+### Web Components
+Web Components are a set of web platform APIs that allow developers to create custom, reusable, and encapsulated HTML elements.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:22]:
```suggestion
Oryx components are build as web components. Web components is a set of web platform standards that allow developers to create custom, reusable, and encapsulated HTML elements.

Oryx uses LIT elements to build web components, but you can use any framework (or plain HTML) to build those components.
```

File: _drafts/oryx/technology.md [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+
+### Web hosting
+Any hosting provider can be used, but SSR specific build will require additional configuration.
+
+### CSR and SSR
+Oryx provides both client-side rendered (CSR) and server-side rendered (SSR) versions of websites, so you can choose the option that best fits your needs.
+
+Client-side rendering involves loading a minimal HTML document and then using JavaScript to build the full page in the browser. This approach is great for creating highly dynamic and interactive user interfaces, as it allows for fast, seamless updates without page refreshes. However, it can negatively impact initial load time and SEO, as search engines may have trouble indexing content.
+
+Server-side rendering, on the other hand, involves generating a complete HTML document on the server and then sending it to the browser. This approach provides faster initial load times, better SEO, and improved accessibility. However, it can be more complex to implement and may not be as effective for highly dynamic pages that require frequent updates.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:32]: We should link into SSR docs, not sure how much of an intro we should add here. Perhaps we make it more slim and add it to the hosting section where you already added a note about SSR?

File: _drafts/oryx/technology.md [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+
+We are  to avoid loading JS in the client unless there's a need; we gradually load snippets of JS rather than a full JS framework.
+
+In the Oryx framework, our approach is to minimize the loading of JavaScript on the client side unless it is necessary. Instead of loading a full JavaScript framework, we opt for gradually loading snippets of JavaScript as required.
+
+### Web Components
+Web Components are a set of web platform APIs that allow developers to create custom, reusable, and encapsulated HTML elements.
+
+### Web hosting
+Any hosting provider can be used, but SSR specific build will require additional configuration.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:25]: We can add some meet to this section: 

Hosting an oryx application would benefit from a couple of capabilities:
- build process and integration into the pipeline
- configuration of http headers (security, caching, etc.)
- CDN integration
- configuration of rewrites and redirects
- integration with lambda functions

File: _drafts/oryx/technology.md [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+
+Client-side rendering involves loading a minimal HTML document and then using JavaScript to build the full page in the browser. This approach is great for creating highly dynamic and interactive user interfaces, as it allows for fast, seamless updates without page refreshes. However, it can negatively impact initial load time and SEO, as search engines may have trouble indexing content.
+
+Server-side rendering, on the other hand, involves generating a complete HTML document on the server and then sending it to the browser. This approach provides faster initial load times, better SEO, and improved accessibility. However, it can be more complex to implement and may not be as effective for highly dynamic pages that require frequent updates.
+
+## Development stack:
+
+### HTML
+HTML is the backbone of the web. It is the markup language used to create web pages and applications. In Oryx, HTML is used to structure the content and layout of the application. It provides a semantic structure to the page, making it easier to understand and maintain.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:37]: We could also add a note regarding accessibility

File: _drafts/oryx/technology.md [hunk 20]
```diff
@@ -0,0 +1,66 @@
+# Oryx technologies
+
+Oryx is a frontend framework that is designed to help developers build web applications using modern technologies.
+When possible and appropriate, Oryx adheres to standards for both technologies and tools.
+Oryx strives to minimize the use of build tools and instead utilize the native capabilities of the web, as much as possible.
+Oryx aims to use JavaScript during development, but tries avoid the need to run JavaScript in the browser. Wherever possible, HTML and CSS are preferred to render a fast loading experience cross different screen sizes.
+
+We will examine both the deployed technology stack and the technologies utilized for development in order to gain a more comprehensive understanding of the technologies.
+
+## Deployed stack:
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:10]: I do like the fact that we articulate deployed tech vs developer tech. But it feels overcomplex and like repetition when you first read it. What about this:
Have a single section (here at the top), describing about how we embrace standard web platform. We do not need to go deep and fine grained. But we do talk a little about web components, avoiding JS in te browser, etc. But we keep it grips, and then have the list of Technologies to build Oryx applications.

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

**Comment by tolerants** [comment]
@tobi tbh I'm a bit lost in the direction... for now Web-hosting section looks out of structure.

File: _drafts/oryx/technology.md [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+
+- Build process integration: Choose hosting providers with seamless build process integration to streamline updates and deployments.
+- HTTP header configuration: Opt for hosting providers that allow flexible configuration of HTTP headers for improved security, caching, and optimizations.
+- CDN integration: Look for hosting providers with native CDN integration to boost performance and availability by distributing static assets across a network.
+- Rewrites and redirects: Prioritize hosting providers that enable easy configuration of URL rewrites and redirects for managing URL structure and routing.
+- Lambda function integration: Select hosting providers that seamlessly integrate with serverless technologies like AWS Lambda to execute custom logic or server-side operations.
+- Consider hosting providers that offer support for [Server-Side Rendering on Oryx](./oryx-server-side-rendering.md). SSR enables dynamic content generation on the server, delivering pre-rendered pages to clients for improved performance, SEO-friendliness, and enhanced user experience.
```
[the remark this answers is not in the document]
Comment (tolerants) [reply _drafts/oryx/technology.md:49]: I've put it into the previous section related to Lamba.

File: _drafts/oryx/technology.md [hunk 22]
```diff
@@ -0,0 +1,51 @@
+# Oryx technologies
+
+Oryx is a frontend framework that is designed to help developers build web applications using modern technologies.
+
+Oryx follows technology and tool standards whenever appropriate. It emphasizes utilizing the inherent capabilities of the web and minimizing the reliance on build tools. For instance, it leverages Web Components to the extent possible. During development, Oryx primarily employs JavaScript but aims to reduce the need for executing JavaScript in the browser. It prioritizes HTML and CSS for delivering a fast-loading experience across various screen sizes.
+
+In terms of styling and layout rendering, Oryx adopts a CSS-first approach, considering CSS as the primary method. It recognizes the inherent capabilities of CSS in handling styling and layout tasks without requiring additional JavaScript logic. Oryx endeavors to avoid loading JavaScript on the client side unless necessary, opting instead to load snippets of JavaScript gradually rather than a full JavaScript framework. This approach minimizes the overall JavaScript payload in the Oryx framework.
```
[the remark this answers is not in the document]
Comment (tolerants) [reply _drafts/oryx/technology.md:7]: I'd say that into is already big enough and this still covered later.

File: _drafts/oryx/technology.md [hunk 23]
```diff
@@ -0,0 +1,67 @@
...
+
+- Build process integration: Choose hosting providers with seamless build process integration to streamline updates and deployments.
+- HTTP header configuration: Opt for hosting providers that allow flexible configuration of HTTP headers for improved security, caching, and optimizations.
+- CDN integration: Look for hosting providers with native CDN integration to boost performance and availability by distributing static assets across a network.
+- Rewrites and redirects: Prioritize hosting providers that enable easy configuration of URL rewrites and redirects for managing URL structure and routing.
+- Lambda function integration: Select hosting providers that seamlessly integrate with serverless technologies like AWS Lambda to execute custom logic, server-side operations or perform a server-side rendering.
+- Consider hosting providers that offer support for [Server-Side Rendering on Oryx](./oryx-server-side-rendering.md). SSR enables dynamic content generation on the server, delivering pre-rendered pages to clients for improved performance, SEO-friendliness, and enhanced user experience.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:65]:
```suggestion

If your Oryx applications requires Consider hosting providers that offer support for [Server-Side Rendering on Oryx](./oryx-server-side-rendering.md) you need to make sure that the right infrastructure is in place at your hosting provider. You can run SSR in a Lambda function or have a long-running server that processes the SSR.

If you have a fairly static application that doesn't require SSR, you could consider a non-frontend hosting provider. 
```

File: _drafts/oryx/technology.md [hunk 24]
```diff
@@ -0,0 +1,67 @@
+# Oryx technologies
+
+Oryx is a frontend framework that is designed to help developers build web applications using modern technologies and leveraging (Spryker API)[https://docs.spryker.com/docs/scos/dev/glue-api-guides/202307.0/decoupled-glue-api.html].
+
+Oryx leverage standard technology, frameworks and tools provided by the web platform and the community wherever possible. It emphasizes utilizing the inherent capabilities of the web and minimizing the reliance on build tools. During development, Oryx primarily employs JavaScript but aims to reduce the need for executing JavaScript in the browser. It prioritizes HTML and CSS for delivering a fast-loading experience across various screen sizes.
+
+Furthermore, Oryx is framework-agnostic and can seamlessly integrate into various web frameworks like React and Vue.js. This versatility is achieved through a combination of Web Components for the user interface and vanilla JavaScript for handling business logic.  This approach offers the flexibility to use the entire library, including its components, or to selectively employ specific layers, such as integration logic.
```
Comment (tobi-or-not-tobi) [inline _drafts/oryx/technology.md:7]:
```suggestion
Furthermore, Oryx is framework-agnostic and can seamlessly integrate into various web frameworks like React and Vue.js. This versatility is achieved through a combination of Web Components for the user interface and vanilla JavaScript for handling business logic. This approach offers the flexibility to use the entire library, including its components, or to selectively employ specific layers, such as integration logic.
```

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

**Review by tobi-or-not-tobi — DISMISSED** [review]

File: _drafts/oryx/technology.md [hunk 25]
```diff
@@ -0,0 +1,70 @@
+# Oryx technologies
+
+Oryx is a frontend framework that is designed to help developers build web applications using modern technologies and leveraging (Spryker API)[https://docs.spryker.com/docs/scos/dev/glue-api-guides/202307.0/decoupled-glue-api.html].
+
+Oryx leverage standard technology, frameworks and tools provided by the web platform and the community wherever possible. It emphasizes utilizing the inherent capabilities of the web and minimizing the reliance on build tools. During development, Oryx primarily employs JavaScript but aims to reduce the need for executing JavaScript in the browser. It prioritizes HTML and CSS for delivering a fast-loading experience across various screen sizes.
```
Comment (andriitserkovnyi) [inline _drafts/oryx/technology.md:5]: Is `wherever possible` related to only `the community` or to all the items in the sentence?

File: docs/scos/dev/front-end-development/202307.0/oryx/technology.md [hunk 26]
```diff
@@ -0,0 +1,77 @@
...
+title: "Oryx: Technology"
+description: The technologies Oryx is built on
+template: concept-topic-template
+last_updated: Oct 23, 2023
+---
+
+Oryx is a frontend framework designed to help developers build web applications using modern technologies and leveraging [Spryker API](https://docs.spryker.com/docs/scos/dev/glue-api-guides/{{page.version}}/decoupled-glue-api.html).
+
+Oryx leverages standard technology, frameworks and tools provided by the web platform and the community wherever possible.
 It emphasizes utilizing the capabilities of the web and minimizing the reliance on build tools. During development, Oryx primarily employs JavaScript (JS) but aims to reduce the need for executing JS in the browser. It prioritizes HTML and CSS for delivering a fast-loading experience across various screen sizes.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202307.0/oryx/technology.md:10]: Is `wherever possible` related to only `the community` or to all the items in the sentence?

File: docs/scos/dev/front-end-development/202307.0/oryx/technology.md [hunk 27]
```diff
@@ -0,0 +1,77 @@
...
+
+## HTML
+
+HTML is a markup language used to create web pages and applications. In Oryx, HTML is used to structure the content and layout of the application. It provides a semantic structure to the page, making it easier to understand and maintain. Also, Oryx prioritizes accessibility principles ensuring access to information and functionality for all users.
+
+## CSS
+
+CSS is a style sheet language used to describe the visual presentation of a document written in HTML.
+
+Oryx uses standard CSS syntax and does not use CSS preprocessors because Oryx does not use a global CSS utility library; instead, we have isolated styles per component and because of that we do not want to load global css utils in each component.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202307.0/oryx/technology.md:36]: Do we need to talk about not using preprocessors or the library?

File: docs/scos/dev/front-end-development/202307.0/oryx/technology.md [hunk 28]
```diff
@@ -0,0 +1,77 @@
...
+You can use Lit to customise Oryx, but you can also use another framework to build web components. Using Lit will give you a few advantages:
+- Reuse Component Mixins and controllers.
+- Out of the box integration with optimised (reactivity)[https://docs.spryker.com/docs/scos/dev/front-end-development/202212.0/oryx/reactivity/reactivity.html] and partial hydration concepts.
+- Out of the box integration with SSR.
+
+## RxJS
+RxJS is a reactive programming library for JS. In Oryx, RxJS is used to handle asynchronous operations. It provides a declarative way to handle events and data streams, making it easier to manage complex data flows and state transitions.
+RxJS is predominantly used in the service layer to provide a reactive system that emit new values while you navigate through the (SPA) experience.
+RxJS is not a main citizen in the component logic, as we try to hide the reactivity altogether to avoid complexity
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202307.0/oryx/technology.md:56]: What is the main citizen? :D

File: docs/scos/dev/front-end-development/202307.0/oryx/technology.md (hunk 26 above)
Comment (tolerants) [reply docs/scos/dev/front-end-development/202307.0/oryx/technology.md:10]: To all items.

File: docs/scos/dev/front-end-development/202307.0/oryx/technology.md (hunk 27 above)
Comment (tolerants) [reply docs/scos/dev/front-end-development/202307.0/oryx/technology.md:36]: I'd say it's nice to have, but not necessary.

File: docs/scos/dev/front-end-development/202307.0/oryx/technology.md (hunk 28 above)
Comment (tolerants) [reply docs/scos/dev/front-end-development/202307.0/oryx/technology.md:56]: It means that we prefer to avoid rxjs usage inside component. We can rephrase it.

File: docs/scos/dev/front-end-development/202307.0/oryx/technology.md [hunk 29]
```diff
@@ -33,7 +33,7 @@ HTML is a markup language used to create web pages and applications. In Oryx, HT
 
 CSS is a style sheet language used to describe the visual presentation of a document written in HTML.
 
-Oryx uses standard CSS syntax and does not use CSS preprocessors because Oryx does not use a global CSS utility library; instead, we have isolated styles per component and because of that we do not want to load global css utils in each component.
+Oryx uses standard CSS syntax. Because Oryx doesn't use a global CSS utility library, it doesn't use CSS preprocessors; instead, styles are isolated per component and global CSS utils are not loaded in them.
```
Comment (tolerants) [inline docs/scos/dev/front-end-development/202307.0/oryx/technology.md:36]: `Oryx uses standard CSS syntax` and `Oryx doesn't use CSS preprocessors` statements logically should be together. So what if we rephrase it:

> Oryx uses standard CSS syntax and does not use CSS preprocessors. Also Oryx doesn't use a global CSS utility library; instead, styles are isolated per component and global CSS utils are not loaded in them.
